### PR TITLE
Global pylintrc file and code quality fixes

### LINKED ISSUE #472: Project-wide pylintrc file
We should agree on and create a `pylintrc` file for the project root.

http://pylint-messages.wikidot.com/all-codes lists all the codes and messages supported by pylint.

Pylint supports disabling all messages and selectively enabling individual ones, or we could just disable the most annoying ones.

### PR DESCRIPTION
**Pull request by achilleas-k**
Added a pylintrc file with the error/warning codes listed in #472 and addressed all the issues it produced.

- Some codes were disabled because they require large code restructuring to be addressed (opened issue #488).
- Some common "bad" names were added to the list of good names, like `da` for DataArray and `mt` for MultiTag.

Closes #472

### CODE REVIEW

**Comment by achilleas-k** [comment]
Travis macOS builds are still timing out.  Not sure what to do about those.

**Comment by achilleas-k** [comment]
Rebased

File: nixio/block.py [hunk 1]
```diff
@@ -228,9 +227,9 @@ def create_data_array(self, name="", array_type="", dtype=None, shape=None,
         if copy_from:
             if not isinstance(copy_from, DataArray):
                 raise TypeError("Object to be copied is not a DataArray")
-            id = self._copy_objects(copy_from, "data_arrays",
+            objid = self._copy_objects(copy_from, "data_arrays",
                                     keep_copy_id, name)
```
Comment (jgrewe) [inline nixio/block.py:231]: actually I am not sure, if I really think that objid is a better variable name than id ;)
``jan.set_nitpick = true``
Anyway, indentation is broken here
``jamn.set_nitpick = false``

File: nixio/dimension_type.py [hunk 2]
```diff
@@ -15,4 +15,3 @@ class DimensionType(Enum):
     Sample = "sample"
     Range = "range"
     Set = "set"
-    DataFrame = "data_frame"
```
Comment (jgrewe) [inline nixio/dimension_type.py:18]: good catch

File: nixio/exceptions/__init__.py [hunk 3]
```diff
@@ -7,5 +7,5 @@
     "DuplicateName", "UninitializedEntity", "InvalidUnit",
     "InvalidAttrType", "InvalidEntity", "OutOfBounds",
     "IncompatibleDimensions", "InvalidFile",
-    "DuplicateColumnName", "UnsuportedLinkType"
```
Comment (jgrewe) [inline nixio/exceptions/__init__.py:10]: How did you find that one, impressive ;)

**Review by jgrewe — APPROVED** [review]

File: nixio/block.py (hunk 1 above)
Comment (achilleas-k) [reply nixio/block.py:231]: `id` is a builtin function in Python so I don't like using it for variable or function names.

File: nixio/exceptions/__init__.py (hunk 3 above)
Comment (achilleas-k) [reply nixio/exceptions/__init__.py:10]: I think the linter caught it :)

**Review by jgrewe — APPROVED** [review]

**Pull request merged**